File: psana_examples/tags/V00-04-13/src/dump_alias.py
#--------------------------------------------------------------------------
# File and Version Information:
#  $Id$
#
# Description:
#  Psana user analysis module dump_alias...
#
#------------------------------------------------------------------------

"""Psana module which dumps alias configuration.

This software was developed for the LCLS project.  If you use all or 
part of it, please give an appropriate acknowledgment.

@version $Id$

@author Andy Salnikov
"""

#------------------------------
#  Module's version from SVN --
#------------------------------
__version__ = "$Revision$"
# $Source$

#--------------------------------
#  Imports of standard modules --
#--------------------------------
import sys
import logging

#-----------------------------
# Imports for other modules --
#-----------------------------
from psana import *

#----------------------------------
# Local non-exported definitions --
#----------------------------------

#---------------------
#  Class definition --
#---------------------
class dump_alias (object) :

    #----------------
    #  Constructor --
    #----------------
    def __init__(self):
        '''Class constructor. Does not need any parameters'''

        self.m_src = self.configSrc('source', 'ProcInfo()')

    #-------------------
    #  Public methods --
    #-------------------

    def beginRun( self, evt, env ) :
        """This optional method is called if present at the beginning 
        of the new run.

        @param evt    event data object
        @param env    environment object
        """

        config = env.configStore().get(Alias.Config, self.m_src)
        if config:
        
            print "%s: %s" % (config.__class__.__name__, self.m_src)
            print "  numSrcAlias =", config.numSrcAlias()
            for i, alias in enumerate(config.srcAlias()):
                print "    {0}: {1} -> {2}".format(i, alias.aliasName(), alias.src())

    def event( self, evt, env ) :
        """This method is called for every L1Accept transition.

        @param evt    event data object
        @param env    environment object
        """

        pass

File: psana_examples/tags/V00-04-16/src/DumpAlias.cpp
//--------------------------------------------------------------------------
// File and Version Information:
// 	$Id$
//
// Description:
//	Class DumpAlias...
//
// Author List:
//      Andy Salnikov
//
//------------------------------------------------------------------------

//-----------------------
// This Class's Header --
//-----------------------
#include "psana_examples/DumpAlias.h"

//-----------------
// C/C++ Headers --
//-----------------

//-------------------------------
// Collaborating Class Headers --
//-------------------------------
#include "MsgLogger/MsgLogger.h"
#include "psddl_psana/alias.ddl.h"

//-----------------------------------------------------------------------
// Local Macros, Typedefs, Structures, Unions and Forward Declarations --
//-----------------------------------------------------------------------

// This declares this class as psana module
using namespace psana_examples;
PSANA_MODULE_FACTORY(DumpAlias)

//		----------------------------------------
// 		-- Public Function Member Definitions --
//		----------------------------------------

namespace psana_examples {

//----------------
// Constructors --
//----------------
DumpAlias::DumpAlias (const std::string& name)
  : Module(name)
  , m_src()
{
  // get the values from configuration or use defaults
  m_src = configSrc("source", "ProcInfo()");
}

//--------------
// Destructor --
//--------------
DumpAlias::~DumpAlias ()
{
}

/// Method which is called at the beginning of the calibration cycle
void 
DumpAlias::beginCalibCycle(Event& evt, Env& env)
{
  shared_ptr<Psana::Alias::ConfigV1> config1 = env.configStore().get(m_src);
  if (config1) {

    WithMsgLog(name(), info, str) {
      str << "Alias::ConfigV1:";
      str << "\n  numSrcAlias = " << config1->numSrcAlias();
      const ndarray<const Psana::Alias::SrcAlias, 1>& srcAlias = config1->srcAlias();
      for (unsigned i = 0; i != srcAlias.size(); ++ i) {
        const Psana::Alias::SrcAlias& alias = srcAlias[i];
        str << "\n    " << i << ": " << alias.aliasName() << " -> " << alias.src() ;
      }
    }

  }
}

/// Method which is called with event data, this is the only required 
/// method, all other methods are optional
void 
DumpAlias::event(Event& evt, Env& env)
{
}
  
} // namespace psana_examples
